core/registration.c

```diff
@@ -1535,7 +1577,7 @@ static lwm2m_client_t * prv_getClientByName(lwm2m_context_t * contextP,
     lwm2m_client_t * targetP;
 
     targetP = contextP->clientList;
-    while (targetP != NULL && strcmp(name, targetP->name) != 0)
+    while (targetP != NULL && (targetP->name == NULL || strcmp(name, targetP->name) != 0))
```

Comment:
Parenthesis

Reply:
Does || has precedence over && ? I think it is the otherway around. Regardless, this makes it clearer. No changes.